src/boot/efi/devicetree.c

```diff
@@ -116,6 +177,12 @@ EFI_STATUS devicetree_install_from_memory(
         if (!state->orig)
                 return EFI_UNSUPPORTED;
 
+        /* Try to match with the firmware-provided DT. If a match is
+         * not found copy the entire .dtb section */
+        err = devicetree_match(&dtb_buffer, &dtb_length, state->orig);
+        if (err != EFI_SUCCESS)
+                log_error_status(err, "No matching device tree found in UKI.");
```

Comment:
No matching dt is not an error here. You ought to log about bad dtbs, though: e.g. `!IN_SET(EFI_SUCCESS, EFI_NOT_FOUND)`

Reply:
So the behavior here should be `if(!IN_SET(err, EFI_SUCCESS, EFI_NOT_FOUND)) { log() }`?